src/Stack/MetricsCollector.php

```diff
@@ -0,0 +1,229 @@
+            [$namespace, $name] = explode('.', $metric);
+
+            switch ($kind) {
+                case 'histogram':
+                    $this->metrics[$metric] = $this->registry->registerHistogram(
+                        $namespace,
+                        $name,
+                        $description,
+                        $labels
+                    );
+                    break;
+
+                case 'counter':
+                    $this->metrics[$metric] = $this->registry->registerCounter(
+                        $namespace,
+                        $name,
+                        $description,
+                        $labels
+                    );
+                    break;
+
+                case 'gauge':
+                    $this->metrics[$metric] = $this->registry->registerGauge(
+                        $namespace,
+                        $name,
+                        $description,
+                        $labels
+                    );
+                    break;
+            }
+        }
+    }
+
+    public function get_gauge($metric)
+    {
+        return $this->metrics[$metric];
+    }
+
+    public function get_counter($metric)
+    {
+        return $this->metrics[$metric];
+    }
+
+    public function get_histogram($metric)
+    {
+        return $this->metrics[$metric];
+    }
+
+    public function render()
+    {
+        $renderer = new \Prometheus\RenderTextFormat();
+        $result = $renderer->render($this->registry->getMetricFamilySamples());
+
+        header('Content-type: ' . \Prometheus\RenderTextFormat::MIME_TYPE);
+
+        echo $result;
+        exit();
+    }
+}
+
+class MetricsCollector
+{
+    const REQUEST_TYPES = array( 'cron', 'api', 'admin-ajax', 'admin', 'search', 'frontpage', 'singular', 'archive', 'other' );
+
+    private $registry;
+    private $metrics;
+    private $wpdb_stats;
+
+    public function __construct()
+    {
+        $this->metrics = new MetricsRegistry(
+            array(
+                'wp.requests'              => array('counter',   'Number of requests',                                 ['request_type']),
+                'wp.page_generation_time'  => array('histogram', 'Page generation time, in seconds',                   ['request_type']),
+                'wp.peak_memory'           => array('histogram', 'Peak memory per request, in bytes',                  ['request_type']),
+                'wpdb.query_time'          => array('histogram', 'Total MySQL query time per request, in seconds',     ['request_type']),
+                'wpdb.num_queries'         => array('histogram', 'Total number of MySQL queries per request',          ['request_type']),
+                'wpdb.num_slow_queries'    => array('histogram', 'Number of MySQL slow queries per request',           ['request_type']),
+                'wpdb.slow_query_treshold' => array('gauge',     'The treshold for counting slow queries, in seconds', []),
```

Comment:
Should this one be considered a metric? I see the point in using it, but maybe we can have it as a label.

Reply:
It's definitely not a label since you canot have a query that is "slow" based on two tresholds, and according to the prometheus docs: `Use labels to differentiate the characteristics of the thing that is being measured`. 
Also they say: 
`CAUTION: Remember that every unique combination of key-value label pairs represents a new time series, which can dramatically increase the amount of data stored. Do not use labels to store dimensions with high cardinality (many different label values), such as user IDs, email addresses, or other unbounded sets of values.`

A single metric that changes rarely (or never) is not going to add too much overhead storage wise. The reason I added it as a metrics is to be able to plot it as treshold in the graphs, without hardcoding it in the grafana dashboard.
By changing the treshold (if it's not stored as a metric) you invalidate all previous data points. 
If it's stored and displayed as a metric, the change will be plotted on the graph, in correlation with the number of slow queries.